packages/runtime/container-runtime/src/opLifecycle/batchManager.ts

```diff
@@ -54,9 +55,14 @@ export class BatchManager {
 
 	constructor(public readonly options: IBatchManagerOptions) {}
 
-	public push(message: BatchMessage, currentClientSequenceNumber?: number): boolean {
+	public push(
+		message: BatchMessage,
+		currentClientSequenceNumber?: number,
+		reentrant: boolean = false,
```

Comment:
Does it need to be optional?

Reply:
Good point. This is the result of my laziness to not have to change all unit tests but it does need to be required, as a forcing function that if/when these things get refactored, we maintain correctness.